skglm/experimental/pdcd_ws.py

```diff
@@ -0,0 +1,219 @@
+import warnings
+
+import numpy as np
+from numpy.linalg import norm
+
+from numba import njit
+from skglm.utils.jit_compilation import compiled_clone
+from sklearn.exceptions import ConvergenceWarning
+
+
+class PDCD_WS:
+    """Primal-Dual Coordinate Descent solver with working sets.
+
+    Solver inspired by [1] that uses working sets [2].
+
+    Parameters
+    ----------
+    max_iter : int, optional
+        The maximum number of iterations or equivalently the
+        the maximum number solved subproblems.
+
+    max_epochs : int, optional
+        Maximum number of CD epochs on each subproblem.
+
+    p0 : int, optional
+        First working set size.
+
+    tol : float, optional
+        The tolerance for the optimization.
+
+    dual_init : array, shape (n_samples,) default None
```

Comment:
For simplicity, given the benchmark results, remove this parameter and initialize at y